src/main/java/spoon/support/sniper/SniperJavaPrettyPrinter.java

```diff
@@ -176,6 +177,46 @@ public void printSourceFragment(SourceFragment fragment, Boolean isModified) {
 	}
 
 
+	/**
+	 * Ugly hack to start printing an arbitrary element of the ast, that is a clone.
+	 * @param element clone to print
+	 * @param original its original
+	 * @return this
+	 */
+	public SniperJavaPrettyPrinter scanClone(CtElement element, CtElement original) {
+		if (element != null) {
+			if (element.getPosition().getCompilationUnit() != null) {
+				CompilationUnit compilationUnit = element.getPosition().getCompilationUnit();
+
+				//use line separator of origin source file
+				setLineSeparator(detectLineSeparator(compilationUnit.getOriginalSourceCode()));
+
+				CtRole role = getRoleInCompilationUnit(original);
+
+				runInContext(new SourceFragmentContextList(mutableTokenWriter,
+								element,
+								Collections.singletonList(original.getOriginalSourceFragment()),
+								new ChangeResolver(getChangeCollector(), element)),
+						() -> {
+							onPrintEvent(new ElementPrinterEvent(role, element) {
+								@Override
+								public void print(Boolean muted) {
+									superScanInContext(element, SourceFragmentContextPrettyPrint.INSTANCE, muted);
+								}
+								@Override
+								public void printSourceFragment(SourceFragment fragment, Boolean isModified) {
+									scanInternal(role, element, fragment, isModified);
+								}
+							});
+						});
+			} else {
+				super.scan(element);
```

Comment:
Such case should be detected in advanced in Element.toString() and we should directly call Element.toStringDebug(), which is safer approach.